homeassistant/components/cloud/account_link.py

```diff
@@ -122,12 +148,15 @@ async def await_tokens() -> None:
                 _LOGGER.info(
                     "Failed to fetch tokens for flow %s: %s", flow_id, err.code
                 )
+            except asyncio.CancelledError:
+                _LOGGER.debug("Token fetch for flow %s was cancelled", flow_id)
+                raise
             else:
                 await self.hass.config_entries.flow.async_configure(
                     flow_id=flow_id, user_input=tokens
                 )
 
-        self.hass.async_create_task(await_tokens())
+        _async_create_task(self.hass, await_tokens())
```

Comment:
I think we can simply make this a background task with `hass.async_create_background_task`.

We may need to update the test to wait for background tasks here:
https://github.com/home-assistant/core/blob/145e02769c7d58c9838178e4448a748c2e616b56/tests/components/cloud/test_account_link.py#L224

Reply:
Add a comment why we need a background task, ie that we want it to cancel on Home Assistant stop, and we don't need to wait for it during setup.

Reply:
A quick search suggests background task errors are dropped on the floor. Also, are background tasks automatically cancelled, or just hard stopped when the process exits?

Up to you, I can do it either way.

Reply:
Expected exceptions in tasks should always be handled inside the task. We do that already here it looks like.

Background tasks are cancelled here when Home Assistant stops:
https://github.com/home-assistant/core/blob/597540b61120a182ad08866a161fd106bf79d2f9/homeassistant/core.py#L1140-L1144

Reply:
Done. I still need to run full test suite on that version.

Reply:
Tests pass except for `tests/components/sms/test_gateway.py`. I can't get `gammu` module to load.

It's unrelated anyway.